Source/DafnyCore/Triggers/QuantifiersCollector.cs

```diff
@@ -59,14 +84,41 @@ protected override bool VisitOneExpr(Expression expr, ref OldExpr/*?*/ enclosing
     }
 
     protected override bool VisitOneStmt(Statement stmt, ref OldExpr/*?*/ st) {
-      if (stmt is ForallStmt) {
-        ForallStmt s = (ForallStmt)stmt;
-        if (s.EffectiveEnsuresClauses != null) {
-          foreach (Expression expr in s.EffectiveEnsuresClauses) {
-            VisitOneExpr(expr, ref st);
-          }
+      if (stmt is ForallStmt { EffectiveEnsuresClauses: { } effectiveEnsuresClauses }) {
+        foreach (var expr in effectiveEnsuresClauses) {
+          VisitOneExpr(expr, ref st);
+        }
+      } else if (stmt is AssignSuchThatStmt { AssumeToken: null } assignSuchThatStmt) {
+        // Create a corresponding exists quantifier. Annoyingly, an ExistsExpr uses BoundVar's whereas the variables
+        // in the AssignSuchThatStmt are IVariable's, so we have to do a substitution back and forth.
+        var substLocalToBoundVar = new Dictionary<IVariable, Expression>();
+        var substBoundVarToLocal = new Dictionary<IVariable, Expression>();
+        var boundVars = new List<BoundVar>();
+        foreach (var localIdentifierExpr in assignSuchThatStmt.GetAssignedLocals()) {
+          var local = localIdentifierExpr.Var;
+          var boundVar = new BoundVar(local.Origin, local.Name, local.Type);
+          var boundVarIdentifierExpr = new IdentifierExpr(boundVar.Origin, boundVar);
+          boundVars.Add(boundVar);
+          substLocalToBoundVar.Add(local, boundVarIdentifierExpr);
+          substBoundVarToLocal.Add(boundVar, localIdentifierExpr);
         }
+
+        var substituterTo = new Substituter(null, substLocalToBoundVar, new Dictionary<TypeParameter, Type>());
+        var existsExpr = new ExistsExpr(assignSuchThatStmt.Origin, boundVars, null,
+          substituterTo.Substitute(assignSuchThatStmt.Expr),
+          new Attributes("_delayTriggerWarning", new List<Expression>(), substituterTo.SubstAttributes(assignSuchThatStmt.Attributes))) {
```

Comment:
It's a bit suspicious that we have to delay trigger-related warnings; is it because of the temporary ExistsExpr that we create?  it would be good to explain why we have to delay the warnings, at least.